app/routes/variantText/variantTexts.js

```diff
@@ -0,0 +1,226 @@
+  return next();
+});
+
+router.route('/:variantText([A-z0-9-]{36})')
+  .get(async (req, res) => {
+    return res.json(req.variantText.view('public'));
+  })
+  .put(async (req, res) => {
+    try {
+      // validate against the model
+      validateAgainstSchema(updateSchema, req.body, false);
+    } catch (error) {
+      const message = `There was an error validating variant text ${error}`;
+      logger.error(message);
+      return res.status(HTTP_STATUS.BAD_REQUEST).json({error: {message}});
+    }
+
+    if (req.body.text) {
+      req.body.text = sanitizeHtml(req.body.text);
+    }
+
+    try {
+      await req.variantText.update(req.body, {userId: req.user.id});
+      await req.variantText.reload();
+      return res.json(req.variantText.view('public'));
+    } catch (error) {
+      logger.error(`Error while trying to update variant text ${error}`);
+      return res.status(HTTP_STATUS.INTERNAL_SERVER_ERROR).json({
+        error: {message: 'Error while trying to update variant text'},
+      });
+    }
+  })
+  .delete(async (req, res) => {
+    try {
+      await req.variantText.destroy();
+      return res.status(HTTP_STATUS.NO_CONTENT).send();
+    } catch (error) {
+      logger.error(`Error while trying to remove variant text ${error}`);
+      return res.status(HTTP_STATUS.INTERNAL_SERVER_ERROR).json({
+        error: {message: 'Error while trying to remove variant text'},
+      });
+    }
+  });
+router.route('/')
+  .get(async (req, res) => {
+    const userProjects = await getUserProjects(db.models.project, req.user);
+    const projectIdents = userProjects.map((project) => {return project.ident;});
+
+    if (req.body.project) {
+      const userHasProjectAccess = projectAccess(req.user, {projects: [{ident: req.body.project}]});
+
+      if (!userHasProjectAccess) {
+        logger.error(`user ${req.user.username} does not have access to variant text ${req.body.project}`);
+        return res.status(HTTP_STATUS.FORBIDDEN).json({
+          error: {message: `user ${req.user.username} does not have access to variant text ${req.body.project}`},
+        });
+      }
+    }
+
+    try {
+      const whereClause = {
+        ...((req.body.templateId == null) ? {} : {templateId: req.body.templateId}),
+        ...((req.body.projectId == null) ? {} : {projectId: req.body.projectId}),
+        ...((req.body.variantName == null) ? {} : {variantName: req.body.variantName}),
+        ...((req.body.variantGkbId == null) ? {} : {variantGkbId: req.body.variantGkbId}),
+        ...((req.body.cancerType == null) ? {} : {cancerType: req.body.cancerType}),
+        ...((req.body.cancerTypeGkbId == null) ? {} : {cancerTypeGkbId: req.body.cancerTypeGkbId}),
+      };
+
+      let results = await db.models.variantText.scope('public').findAll({
+        where: whereClause,
+      });
+      if (!isAdmin(req.user)) {
+        results = results.filter((variantText) => {return projectIdents.includes(variantText.project.ident);});
```

Comment:
add a check here for the all-projects-access role (or if that role hasn't been implemented yet, a todo with a ticket number to add handling for the all-projects-access role)

Reply:
CREATE_DATA and UPLOAD_DATA are both modified in line 94 to 98 to be different, since uploading requires an Ident, and creating directly an ID

Reply:
the getUserProjects that was used in line 153 already checks for hasAllProjectsAccess. So that should be working already, would you like me to add a test to make sure its working as intended?

Reply:
Yes please re a test. If a user is not admin, and does have all-project access, we do want them to get results from all projects. AFAICT not all projects will necessarily be included in projectIdents (since this is from 151/152, looking for specific project membership and not checking user roles). I think this will only affect results that don't specify what project the variant should have and where the user has the all-project access role.

Reply:
As discussed offline, the function is working as intended, resolving comment